src/Task/Deptrac.php

```diff
@@ -54,11 +70,47 @@ public function run(ContextInterface $context): TaskResultInterface
 
         $arguments = $this->processBuilder->createArgumentsForCommand('deptrac');
         $arguments->add('analyze');
-        $arguments->add('--formatter-graphviz='.(int) $config['formatter_graphviz']);
-        $arguments->add('--formatter-graphviz-display='.(int) $config['formatter_graphviz_display']);
-        $arguments->addOptionalArgument('--formatter-graphviz-dump-image=%s', $config['formatter_graphviz_dump_image']);
-        $arguments->addOptionalArgument('--formatter-graphviz-dump-dot=%s', $config['formatter_graphviz_dump_dot']);
-        $arguments->addOptionalArgument('--formatter-graphviz-dump-html=%s', $config['formatter_graphviz_dump_html']);
+
+        if (empty($config['formatter']) === true) {
+            // Deprecated Deptrac versions < v0.9.0
+            $arguments->add('--formatter-graphviz='.(int) $config['formatter_graphviz']);
+            $arguments->add('--formatter-graphviz-display='.(int) $config['formatter_graphviz_display']);
+            $arguments->addOptionalArgument(
+                '--formatter-graphviz-dump-image=%s',
+                $config['formatter_graphviz_dump_image']
+            );
+            $arguments->addOptionalArgument(
+                '--formatter-graphviz-dump-dot=%s',
+                $config['formatter_graphviz_dump_dot']
+            );
+            $arguments->addOptionalArgument(
+                '--formatter-graphviz-dump-html=%s',
+                $config['formatter_graphviz_dump_html']
+            );
+        }
+
+        // Deptrac versions >= v0.9.0
+        $arguments->addArgumentArray('--formatter=%s', $config['formatter']);
+
+        if (in_array('graphviz', $config['formatter'], true) === true) {
```

Comment:
Would it make sense to normalize the input options instead of adding all these in_arrays?
E.g.: if graphiz is not included in the formatter, the graphiz config option is set to null?

That would keep the task flow a bit more straight forward..

Reply:
I now see that  `addOptionalArgument` is actually doing what was needed :). Updated!